### PR TITLE
Setup github actions

### PR DESCRIPTION
**Pull request by fisker**
<a href="https://github.com/fisker/prettier/actions">
    Github Actions Build Status</a>

  <a href="https://github.com/fisker/prettier/actions">
    Github Actions Build Status</a>

  <a href="https://github.com/fisker/prettier/actions">
    Github Actions Build Status</a>

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by lipis** [comment]
#7401

**Comment by fisker** [comment]
I setup this for  test , seems not too much difference on speed.  

https://github.com/fisker/prettier/actions

**Comment by fisker** [comment]
Also it's not running  on this PR, is it because the org account has no actions permission?

**Comment by fisker** [comment]
Anyone know how to make this run?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - next
+      - git-actions
+  pull_request:
```
Comment (lipis) [inline .github/workflows/ci.yml]:
```
on: [pull_request]
```

should be enough..

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        run: yarn check-types
+
+      - name: Lint code
+        run: yarn lint
+
+      - name: Lint docs
+        run: yarn lint-docs
+
+      - name: Spellcheck
+        run: yarn spellcheck
```
Comment (lipis) [inline .github/workflows/ci.yml]: can we separate `lint.yml` and `test.yml`?

**Comment by lipis** [comment]
Strange that they don't appear, can you add the workflows section in the end.. something like that:

```
workflows:
  version: 2
  lint:
    jobs:
      - lint
```

File: .github/workflows/dev-test.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Dev_Test
+
+on: [pull_request]
```
Comment (lipis) [inline .github/workflows/dev-test.yml]: maybe on `[push]`?

I'm runnuing out of ideas.. maybe we can merge it and see what happens.

**Comment by lipis** [comment]
Merged it, so maybe it will trigger them.. if it works let's start adjusting those.

**Comment by fisker** [comment]
Great, running

**Comment by lipis** [comment]
Now let's optimize them and move more here.. and start removing the duplicate ones...

**Comment by lipis** [comment]
I just noticed that we did that on `next` branch... why not on master?

**Comment by lipis** [comment]
Can you create a new PR for master and sync `next` afterwards?

**Comment by fisker** [comment]
Scripts are different from `master` and `next`, why waste time write actions config for old scripts?

**Comment by lipis** [comment]
fair enough... we should soon be done with 2.0 anyways.

**Pull request merged**